src/components/data/QueryAuth0.js

```diff
@@ -37,24 +38,82 @@ export default class QueryAuth0 extends React.PureComponent {
       authenticationFailed(err);
     }
 
+    // By default this is true if the window.location.hash parsing stuff worked.
+    // We might change our minds about this later.
+    let startAccessTokenRefreshLoop = !!authResult;
+
     if (!authResult) {
       // It if was null from finishAuthenticationFlow() it means it didn't pick anything
       // up from the window.location.hash so we can potentially use what was stored in
       // localStorage.
       authResult = JSON.parse(localStorage.getItem('authResult'));
+      if (authResult) {
```

Comment:
this could just be:
```js
startAcessTokenRefreshLoop = !!authResult;
```

Reply:
I don't see that. We need to evaluate the value of `authResult` **twice**. 
If it was something true'y after the `parseHash` we should start the refresh loop. But also, if it was falsy after `parseHash` and we *did* have something in `localStorage` that should also start the refresh loop. 

Can you take a second look? Perhaps my code comments weren't clear.

Reply:
ah! my bad. I didn't think about the fact that this was already assigned a value above.

this could be:
```js
startAccessTokenRefreshLoop = authResult ? true : startAccessTokenRefreshLoop;
```